src/main/java/Duke.java

```diff
@@ -1,10 +1,163 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final int MAX_TASKS = 100;
+
+    private static final Task[] tasks = new Task[MAX_TASKS];
+
+    private static int counter = 0;
+
+    private static boolean isRunning = true;
+
+    /**
+     * Prints out all the tasks in the tasks array
+     */
+    private static void printTasks() {
+        System.out.println(Message.LIST_TASKS.message);
+        for (int i = 0; i < counter; ++i) {
+            System.out.printf("%d. %s\n",
+                    (i + 1),
+                    tasks[i].getDescriptionText()
+            );
+        }
+    }
+
+    /**
+     * Adds a todo task into the array and prints out the
+     * corresponding message
+     *
+     * @param taskDescription A string after removing the
+     *                        command
+     */
+    private static void addTodoTask(String taskDescription) {
+        Todo newTodo = new Todo(taskDescription);
+        tasks[counter] = newTodo;
+        counter += 1;
+        System.out.printf(
+                Message.TASK_ADDED.message,
+                newTodo.getDescriptionText(),
+                counter
+        );
+    }
+
+    /**
+     * Adds a deadline task into the array and prints out
+     * the corresponding message
+     *
+     * @param taskDescription A string after removing the
+     *                        command
+     */
+    private static void addDeadlineTask(String taskDescription) {
+        int index = taskDescription.indexOf("/by");
+        String description = taskDescription.substring(0, index);
+        String endDuration = taskDescription.substring(index + "/by ".length());
+        Deadline newDeadline = new Deadline(description, endDuration);
+        tasks[counter] = newDeadline;
+        counter += 1;
+        System.out.printf(
+                Message.TASK_ADDED.message,
+                newDeadline.getDescriptionText(),
+                counter
+        );
+    }
+
+    /**
+     * Adds an event task into the tasks array and prints
+     * out the corresponding message
+     *
+     * @param taskDescription A string after removing the
+     *                        command
+     */
+    private static void addEventTask(String taskDescription) {
+        int indexOfFrom = taskDescription.indexOf("/from");
```

Comment:
perhaps it would be tidier to refactor "/from" into a constant, since "/from" is used again in line 77.